ocp_resources/virtual_machine.py

```diff
@@ -292,6 +292,11 @@ def wait_for_vmi_condition_pause_status(self, pause, timeout=TIMEOUT):
             func=self.get_vmi_active_condition,
         )
         for sample in samples:
+            # VM in state change
+            if (pause and not sample.get("reason")) or (
+                sample.get("reason") == "PausedByUser" and not pause
+            ):
+                continue
```

Comment:
Sorry, but I don't understand what should be the results here
Can you please add comments for each and explain when and why this is needed?

maybe somthing like:
xxx = pause and not sample.get("reason")  # explain ......
yyy = sample.get("reason") == "PausedByUser"  # explain ......

and please elaborate more about "# VM in state change"

Reply:
Comment updated